include/open62541pp/types.hpp

```diff
@@ -875,6 +875,21 @@ enum class VariantPolicy {
     // clang-format on
 };
 
+/**
+ * Token used for selecting the policy when constructing variants.
+ *
+ * TODO:
```

Comment:
The STL uses tag types always as the first constructor argument, for example `std::pair` with `std::piecewise_construct_t` or `std::optional` with `std::in_place_t`. We should do that as well.
The default should always be the safe choice which is copying. Therefore we need tag types for referencing / borrowing the provided value.

Naming is hard, but this would be my favorites:
```cpp
struct BorrowTag {};
inline constexpr BorrowTag borrow{};
```

Other naming ideas are:
- `NoCopyTag`
- `NoDeleteTag`

What do you think?

Reply:
Indeed naming is hard :D
I have no better idea than what you have already chosen for the variant policy. I don't understand the other names yet (are they all alternatives to a potential `ReferenceTag`?):

- `BorrowTag` makes me think of an actual borrow checker, which we don't have in C++ (yet ^^)
- `NoCopyTag` could mean referencing the data or moving it into the Variant
- `NoDeleteTag` excludes ownership and suggests no copy but seems very indirect

Reply:
Outcome of discussion: Let's use `ReferenceTag`

Reply:
Now only `ReferenceTag` and an instance of it `reference` is left.